node/cli/src/chain_spec.rs

```diff
@@ -143,15 +143,30 @@ pub fn staging_testnet_config() -> ChainSpec {
 	)
 }
 
-fn testnet_genesis(initial_authorities: Vec<AuthorityId>, upgrade_key: AccountId) -> GenesisConfig {
-	let endowed_accounts = vec![
-		ed25519::Pair::from_seed(b"Alice                           ").public().0.into(),
-		ed25519::Pair::from_seed(b"Bob                             ").public().0.into(),
-		ed25519::Pair::from_seed(b"Charlie                         ").public().0.into(),
-		ed25519::Pair::from_seed(b"Dave                            ").public().0.into(),
-		ed25519::Pair::from_seed(b"Eve                             ").public().0.into(),
-		ed25519::Pair::from_seed(b"Ferdie                          ").public().0.into(),
-	];
+/// Helper function to generate AuthorityID from seed
+pub fn get_authority_id_from_seed(seed: &str) -> AuthorityId {
+	// NOTE from ed25519 impl:
+	// prefer pkcs#8 unless security doesn't matter -- this is used primarily for tests.
+	let mut padded_seed: [u8; 32] = Default::default();
```

Comment:
Please avoid making breaking changes like this. Seeds that are less than 32 bytes must be padded with spaces or generation from seed here will be incompatible with generation in other tools (e.g. in UI, CLI and `subkey`).

> unless security doesn't matter

I see no reason whatsoever that switching padding from bytes of value `0x20` bytes to value `0x00` will have any impact on security.

Reply:
sorry about the breaking change. i thought i verified, but obviously overlooked. should be fixed and tested.

regarding the security bit, i think the comment may have been confusing because of placement. i just copied over the [comment](https://github.com/paritytech/substrate/blob/101d673cd42bbaa7a56eafd3d60a8598cb7af46d/core/primitives/src/ed25519.rs#L216) from the `ed25519` impl to make it clear that this was potentially sub-optimal and intended for testing.